### ISSUE TITLE
去模糊特徵圖R的可視化圖

### ISSUE THREAD

**Issue by joe880923** [megvii-research/DCLS-SR#49]
作者您好，想請問你可以提供去模糊特徵圖R的可視化方法嗎?
有嘗試使用一些方法但可視化的圖片很奇怪。

**Comment by Algolzw** [comment]
你好，可以尝试使用matplotlib并设置`cmap='jet'` .

**Comment by joe880923** [comment]
作者您好，我儲存R特徵圖的code是這樣
        test = self.channel_1(f2)  #將R特徵圖channel做1*1con讓channel為1
        test = np.squeeze(test, axis=2)
        ndarr = test.detach().squeeze().cpu().numpy()
        ndarr = (ndarr * 255).clip(0, 255).astype(np.uint8)
        print("ndarr-------------------", ndarr.shape)
        print("ndarr dtype-------------------", ndarr.dtype)
        print("ndarr min/max-----------------", ndarr.min(), ndarr.max())
        ndarr = ndarr.astype(np.uint8)
        ndarr_transposed = np.transpose(ndarr, (0, 1))  # 对于二维数组，不需要改变轴的顺序

        plt.imshow(ndarr_transposed, cmap='jet')
        plt.savefig('feature_map.png')
        plt.show()


儲存的結果長這樣是正確的嗎?
![image](https://github.com/megvii-research/DCLS-SR/assets/54349027/4fcd2841-0e4b-4fe7-bf48-b0b2313d91cb)

![image](https://github.com/megvii-research/DCLS-SR/assets/54349027/2a092c28-b48e-490b-b27c-391db97ee8a1)

![image](https://github.com/megvii-research/DCLS-SR/assets/54349027/f319889a-cb1b-495a-97f1-bebdb974b7e1)

感覺應該正常不會長這樣，再麻煩您跟我說一下，謝謝!

**Comment by Algolzw** [comment]
看起来不对= =。你为什么要加一个1x1卷积让channel为1呢？可以直接只取第一个(或者前N个)通道的特征不？

**Comment by joe880923** [comment]
我原本的理解是如果只使用第一個channel的話會導致資訊流失，依照您給的建議有嘗試過使用第1個channel來測試，如下:
        first_channel = f2[0, :1, :, :]
        test = np.squeeze(first_channel, axis=0)
        ndarr = test.detach().squeeze().cpu().numpy()
        ndarr = (ndarr * 255).clip(0, 255).astype(np.uint8)
        plt.imshow(ndarr, cmap='jet')
        plt.savefig('feature_map.png')
        plt.show()
        
印出的結果為
![image](https://github.com/megvii-research/DCLS-SR/assets/54349027/4b19a468-7f7e-4d2e-b2ab-0b1bc7f52b65)

![image](https://github.com/megvii-research/DCLS-SR/assets/54349027/483d8523-d581-4be1-8db0-f530db366242)

好像更加的不太對的樣子，不太確定我哪邊理解有問題或操作，再麻煩您給我點指導，謝謝您!

**Comment by Algolzw** [comment]
你这是学习后的模型吗还是random weights？可以多打印几个feature通道看看呢，理论上有些通道信息比较多。

**Comment by joe880923** [comment]
作者您好，我是使用已經訓練完後的權重檔案去印出圖片，作者您說的多印幾個是指我針對第一個通道、第二個通道、第三個通道打印出來分別儲存起來的意思嗎?

**Comment by Algolzw** [comment]
是的，有的通道信息应该会多一点
